backend/src/services/premium/enrichment/organizationEnrichmentService.ts

```diff
@@ -0,0 +1,216 @@
+        enrichedOrganizations.push({ ...org, id: instance.id, tenantId: this.tenantId })
+        enrichedCacheOrganizations.push({ ...org, id: instance.cachId })
+      }
+    }
+    const orgs = await this.update(enrichedOrganizations, enrichedCacheOrganizations)
+    await this.sendDoneSignal(orgs)
+    return orgs
+  }
+
+  private async update(orgs: IOrganizations, cacheOrgs: IOrganizations): Promise<IOrganizations> {
+    try {
+    await OrganizationCacheRepository.bulkUpdate(cacheOrgs, this.options)
+    return await OrganizationRepository.bulkUpdate(orgs, this.fields, this.options)
+      
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log(error)
+    }
+    return []
+  }
+
+  private convertEnrichedDataToOrg(data: Awaited<IEnrichmentResponse>): IOrganization {
+    data = renameKeys(data, {
+      summary: 'description',
+      employeeCountByCountry: 'employee_count_by_country',
+      twitter_url: 'twitter',
+    })
+    const lastEnrichedAt = new Date()
+    const location = `
+      ${data.location.street_address} ${data.location.address_line_2} ${data.location.name}
+    `
+    const org: IOrganization = lodash.pick(data, this.fields)
+
+    return Object.assign(org, { lastEnrichedAt, location })
+  }
+
+  private async sendDoneSignal(organizations: IOrganizations) {
+    const redis = await createRedisClient(true)
+    const organizationIds = organizations.map((org) => org.id)
+
+    const apiPubSubEmitter = new RedisPubSubEmitter('api-pubsub', redis, (err) => {
+      this.log.error({ err }, 'Error in api-ws emitter!')
+    })
+    if (!organizations.length) {
+      apiPubSubEmitter.emit(
+        'user',
+        new ApiWebsocketMessage(
+          'organization-bulk-enrichment',
+          JSON.stringify({
+            tenantId: this.tenantId,
+            success: false,
+            organizationIds,
+          }),
+          undefined,
+          this.tenantId,
+        ),
+      )
+    }
+    // Send success message if there were enrichedMembers
+    else {
+      apiPubSubEmitter.emit(
+        'user',
+        new ApiWebsocketMessage(
+          'organization-bulk-enrichment',
+          JSON.stringify({
+            tenantId: this.tenantId,
+            success: true,
+            organizationIds,
+          }),
+          undefined,
+          this.tenantId,
+        ),
+      )
+    }
+  }
+
+  private selectFieldsForEnrichment(org: IEnrichableOrganization): IEnrichableOrganization {
+    this.fields.push('name', 'lastEnrichedAt')
+    if(OrganizationEnrichmentService.shouldReenrich(org)) {
+      return org
+    }
+    for(const field of Object.keys(org)) {
+      if(org[field] === null){
+        this.fields.push(field)
+      }
+    }
+    return lodash.pick(org, ['id', 'cachId', ...this.fields])
+  }
+
+  async queryTenancyOrganizations(): Promise<IEnrichableOrganization[]> {
+    const options = await SequelizeRepository.getDefaultIRepositoryOptions()
+    const query = `
```

Comment:
Same as before, let's keep queries in the repo layer. Feel free to use organizations repo.

Reply:
Ditto